core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -246,18 +252,77 @@ sealed trait Message[F[_]] extends Media[F] { self =>
   override def covary[F2[x] >: F[x]]: SelfF[F2] = this.asInstanceOf[SelfF[F2]]
 
   /** Compiles the body stream to a single chunk and sets it as the
-    * body.  Replaces any `Transfer-Encoding: chunked` with a
-    * `Content-Length` header.  It is the caller's responsibility to
-    * assure there is enough memory to materialize the body.
+    * body. Replaces any `Transfer-Encoding: chunked` with a
+    * `Content-Length` header. It is the caller's responsibility to
+    * assure there is enough memory to materialize the body. Also,
+    * potentially this method could run infinitely.
+    * Consider to use [[org.http4s.Message.toStrict]] instead.
     */
-  def toStrict(implicit F: Concurrent[F]): F[Self] =
+  def unsafeToStrict(implicit F: Concurrent[F]): F[Self] =
```

Comment:
I'm confused what makes this unsafe.  We deal with effects that never complete all the time.  There is a risk of an infinite stream, but we're using methods fs2 doesn't consider unsafe.  And I think this is equivalent to the "safe" `unsafeToStrict(Duration.Inf, None)`.

Reply:
I supposed when implementing that the main problem we defining the public method, it's better if users will pay more attention when using this method. Also, `akka-http` have the same [method](https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/basic-directives/toStrictEntity.html) with time/size limitations.